### PR TITLE
Add support for file-based blocks and embeds/bookmarks

### PR DESCRIPTION
**Pull request by duanejeffers**
Hi. Random PR here, but I am needing support for these block types for my integration that I am working on.

This PR should bring your codebase in line with the documentation for blocks.

The only real issue I had when doing this is what to call the File object that is attached to the blocks, so, I went with `BlockFile` but, obviously, that breaks naming conventions. If you have a better suggestion, or if you wanted to rename the File struct that is used by the Property, I'll leave that up to you.

Please let me know if there is anything else that I can do to help. :-D

### CODE REVIEW

File: block.go [hunk 1]
```diff
@@ -273,6 +273,128 @@ func (b ChildPageBlock) GetType() BlockType {
...
+}
+
+type ImageBlock struct {
+	Object         ObjectType `json:"object"`
+	ID             BlockID    `json:"id,omitempty"`
+	Type           BlockType  `json:"type"`
+	CreatedTime    *time.Time `json:"created_time,omitempty"`
+	LastEditedTime *time.Time `json:"last_edited_time,omitempty"`
+	HasChildren    bool       `json:"has_children,omitempty"`
+	Image          Image      `json:"image,omitempty"`
```
Comment (jomei) [inline block.go:302]: Why `omitempty` is used here? As I understood, this prop can't be empty.

**Review by jomei — COMMENTED** [review]
Hi, @duanejeffers! Great job!  A few questions below

File: block.go [hunk 2]
```diff
@@ -273,6 +273,128 @@ func (b ChildPageBlock) GetType() BlockType {
...
+}
+
+func (b ImageBlock) GetType() BlockType {
+	return b.Type
+}
+
+type Image struct {
+	Caption  []RichText `json:"caption,omitempty"`
+	Type     FileType   `json:"type,omitempty"`
+	File     FileObject `json:"file,omitempty"`
```
[the remark this answers is not in the document]
Comment (duanejeffers) [reply block.go]: Great catch. I got that fixed on all the structs where that is used.

File: block.go (hunk 1 above)
Comment (duanejeffers) [reply block.go]: Yep. I got that fixed across all the structs as well. Thanks

**Comment by duanejeffers** [comment]
Thank you very much @jomei !

**Pull request merged**